src/testing/deploy/determined_test.go

```diff
@@ -0,0 +1,197 @@
+//go:build k8s
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/auth"
+	"github.com/pachyderm/pachyderm/v2/src/internal/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/minikubetestenv"
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+	"github.com/pachyderm/pachyderm/v2/src/internal/testutil"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"github.com/pachyderm/pachyderm/v2/src/pps"
+)
+
+const (
+	detLoginPath       = "/api/v1/auth/login"
+	detUserPath        = "/api/v1/users"
+	detWorkspacePath   = "/api/v1/workspaces"
+	detNewUserPassword = "test-password"
+)
+
+type DeterminedUser struct {
+	Id          int    `json:"id"`
+	Username    string `json:"username"`
+	Admin       bool   `json:"admin"`
+	Active      bool   `json:"active"`
+	DisplayName string `json:"displayName"`
+}
+type DeterminedUserBody struct {
+	User     *DeterminedUser `json:"user"`
+	Password string          `json:"password"`
+	IsHashed bool            `json:"isHashed"`
+}
+type DeterminedUserList struct {
+	Users *[]DeterminedUser `json:"users"`
+}
+
+func TestDeterminedInstallAndIntegration(t *testing.T) {
```

Comment:
this looks great! A couple cases we could consider automating are when a user creates a pipeline against a workspace that (A) doesn't exist, or (B) the user doesn't have access to